src/main/java/org/cascadebot/cascadebot/CascadeBot.java

```diff
@@ -196,6 +198,17 @@ private void init() {
             );
         }
 
+        databaseManager.runAsyncTask(database -> {
+            ChangeStreamIterable<GuildData> changeStreamIterable = database.getCollection(GuildDataManager.COLLECTION, GuildData.class).watch();
+            new Thread(() -> changeStreamIterable.forEach(guildDataChangeStreamDocument ->
+                    GuildDataManager.replaceInternal(guildDataChangeStreamDocument.getFullDocument()), (result, throwable) -> {
+                if (throwable != null) {
+                    throwable.printStackTrace();
+                }
+                System.out.println(result);
+            }), "mongoChangeWatch").start();
```

Comment:
Do we need to start this in a separate thread? Since it's a callback I would have assumed it was non-blocking to be honest

Reply:
Good question, it is a foreach which is usually blocking, but we're using the async client (and maybe the not async also is no blocking), I'll have to test this.